README.md

```diff
@@ -21,15 +21,56 @@ rebuilds should be very fast.
 
 ## Adding new packages
 
+**WARNING!** To do this you will need:
+ * Enough bandwidth to comfortably push potentionally gigabytes worth of
+   Docker layers.
+ * Several hours worth of CPU time to re-compile all the packages (if
+   this is the first time you've done this and don't have any layers
+   cached).
+ * Push access to ghcr.io.
+
+If you don't have all these things then please don't start.
+
+### Install it
+
 To add a package, it must be available on CRAN. We cannot currently install
 things from Github or other locations.
-
-`just add-package PACKAGE` 
+```
+just add-package PACKAGE
+```
 
 This will attempt to install and build the package and its dependencies, and
 update the `renv.lock`. It will then rebuild the R image with the new lock file
 and test it.
 
+Note that the first time you do this it will need to compile every
+included R package (because you won't have any Docker layers cached
```

Comment:
```suggestion
included R package (because you won't have the R package builds cached
```

Reply:
Thanks, that a useful correction to my mental model.